### PR TITLE
Waypoint updater and twist controller

### PR DESCRIPTION
**Pull request by scs50**
This PR allows the vehicle to follow waypoints at a set velocity. It does this by modifying the waypoint_updater.py to find the closest waypoints in front of the vehicle and control is provided to the vehicle by the twist_controller.py. Currently does not take into account any traffic lights or obstacles.

Run this by launching launch/styx.launch. Make sure to comment out the traffic light launch files. Once it is running start the simulator and uncheck the box for the manual drive.

### CODE REVIEW

**Comment by manu291** [comment]
I have tested the code. It works very well until the car gets near the end of its first lap, at the 50 mph speed limit traffic sign (possibly at the last way point). Then the ROS process dies with that message:
```bash
Traceback (most recent call last):
  File "/media/sf_udacity/CarND-Capstone-System-Integration/ros/src/twist_controller/dbw_node.py", line 176, in <module>
    DBWNode()
  File "/media/sf_udacity/CarND-Capstone-System-Integration/ros/src/twist_controller/dbw_node.py", line 90, in __init__
    self.loop()
  File "/media/sf_udacity/CarND-Capstone-System-Integration/ros/src/twist_controller/dbw_node.py", line 119, in loop
    self.twist, self.velocity, dt.to_sec())
  File "/media/sf_udacity/CarND-Capstone-System-Integration/ros/src/twist_controller/twist_controller.py", line 47, in control
    brake_torque = self.acceleration_to_torque(abs(self.decel_limit))
AttributeError: 'Controller' object has no attribute 'acceleration_to_torque'
[dbw_node-4] process has died [pid 4076, exit code 1, cmd /media/sf_udacity/CarND-Capstone-System-Integration/ros/src/twist_controller/dbw_node.py __name:=dbw_node __log:=/home/student/.ros/log/1222266c-3655-11e8-809d-080027eb5237/dbw_node-4.log].
log file: /home/student/.ros/log/1222266c-3655-11e8-809d-080027eb5237/dbw_node-4*.log

```
Here are screen shots just before and after the ROS process dies.
screenshot from 2018-04-02 11-07-22
screenshot from 2018-04-02 11-03-24

To reproduce the crash:
First drive manually. Change direction and drive about 200 meters back before the starting position (get past the speed limit sign). Then turn off manual mode.

By the way, congrats for your contribution! You've done it quickly and it works impressively well and the bug mentioned above is probably just a detail to fix.

File: ros/src/twist_controller/twist_controller.py [hunk 1]
```diff
@@ -1,14 +1,63 @@
...
+        steer = self.yaw_controller.get_steering(self.cmd_vel,
+                                                 self.cmd_ang_vel,
+                                                 self.veh_vel)
+
+        # Apply low-pass filter to the linear acceleration
+        cmd_acc = self.velocity_filter.filt(cmd_acc)
+        # Keep full brake if target velocity is almost 0
+        if self.cmd_vel < FULL_BRAKE_SPEED:
+            throttle = 0.0
+            brake_torque = self.acceleration_to_torque(abs(self.decel_limit))
```
Comment (manu291) [inline ros/src/twist_controller/twist_controller.py]: The method `self.acceleration_to_torque` is not implemented which is the reason of the crash when the car finishes the lap.
However it seems that everything works fine if we always use the `else` branch of the test.

Comment (scs50) [reply ros/src/twist_controller/twist_controller.py]: OK, i had a function to convert the accel/decel to torque but I ended up removing it, I will adjust the brake torque in the if statement. What is the best way to change this? New pull request?

Comment (Labonneguigue) [reply ros/src/twist_controller/twist_controller.py]: I believe, you can amend the commit cff49d6 and push again.

Comment (manu291) [reply ros/src/twist_controller/twist_controller.py]: It is recommended to not amend commits that have been already pushed. It is better to do a new commit on top of the branch and then push the branch again.

**Comment by Labonneguigue** [comment]
This branch needs rebasing on master to solve the merge conflict

**Review by manu291 — APPROVED** [review]

**Pull request merged**